tests/test_plugin.py

```diff
@@ -555,3 +562,66 @@ def test_plugin_alias(target_bare: Target) -> None:
     assert target_bare.has_function("bar")
     assert target_bare.has_function("baz")
     assert list(target_bare.foo()) == list(target_bare.bar()) == list(target_bare.baz())
+
+
+@pytest.mark.parametrize(
+    "func_path, func",
+    [(func.path, func) for func in find_plugin_functions(Target(), "*", compatibility=False, show_hidden=True)[0]],
+)
+def test_exported_plugin_format(func_path: str, func: PluginFunction) -> None:
+    """This test checks plugin style guide conformity for all exported plugins.
+
+    Resources:
+        - https://docs.dissect.tools/en/latest/contributing/style-guide.html
+    """
+
+    # Ignore DefaultPlugin and NamespacePlugin instances
+    if func.class_object.__base__ is NamespacePlugin or func.class_object is DefaultPlugin:
+        return
+
+    # Plugin method should specify what it returns
+    assert func.output_type in ["record", "yield", "default", "none"]
+
+    py_func = getattr(func.class_object, func.method_name)
+    annotations = None
+
+    if hasattr(py_func, "__annotations__"):
+        annotations = py_func.__annotations__
+
+    elif isinstance(py_func, property):
+        annotations = py_func.fget.__annotations__
+
+    # Plugin method should have a return annotation
+    assert annotations and "return" in annotations.keys(), f"No return type annotation for function {func}"
+
+    # TODO: Check if the annotations make sense with the provided output_type
+
+    # Plugin method should have a docstring
+    method_doc_str = py_func.__doc__
+    assert isinstance(method_doc_str, str)
+    assert method_doc_str != ""
+
+    # The method docstring should compile to rst without warnings
+    try:
+        publish_string(method_doc_str, settings_overrides={"halt_level": 2})
+
+    except SystemMessage as e:
+        # Limited context was provided to docutils, so some exceptions could incorrectly be raised.
+        # We can assume that if the rst is truly invalid this will also be caught by `tox -e build-docs`.
+        if "Unknown interpreted text role" not in str(e):
+            assert str(e) in method_doc_str  # makes reading pytest error easier
+
+    # Plugin class should have a docstring
+    class_doc_str = func.class_object.__doc__
+    assert isinstance(class_doc_str, str)
+    assert class_doc_str != ""
+
+    # The class docstring should compile to rst without warnings
```

Comment:
Maybe put this into an `assert_valid_rst` function? Because I think down the line (not as part of this PR) we may want to put some more "heuristic" checks in there as well. For example, if a line that's expected to end with a period ends with one, or use of "Resource/Reference" over "References". I'll create a ticket today for this.